lib/config.ts

```diff
@@ -186,7 +185,7 @@ function dedupeProjectName(projectNames: Set<string>, projectName: string) {
  * - an array of valid, deduped projects
  * - the `variants` and `format` config options
  */
-function parseSourceInformation(file?: string) {
+function parseSourceInformation(file?: string): SourceInformation {
```

Comment:
Aren't functions usually best left untyped? Why do we need the explicit type here?

Reply:
In this case, the explicit type is helpful: when we make changes to `SourceInformation`, we want the type errors to show up here instead of showing up downstream where the implicit return value of this function is used. 

When they show up downstream, we end up having to trace them to this function anyway, so the explicit return definition is helpful for developer experience.